frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts

```diff
@@ -383,15 +383,29 @@ type KeyValuePair = {
   key: string;
   value: string;
 };
-export const getSecretAnnotations = (annotation: KeyValuePair) => {
-  const annotations = {};
+
+const getAnnotationKey = (secretType: string, suffix: number) => {
   const annotationPrefix = 'tekton.dev';
-  if (annotation?.key === SecretAnnotationId.Git) {
-    annotations[`${annotationPrefix}/${SecretAnnotationId.Git}-0`] = annotation?.value;
-  } else if (annotation?.key === SecretAnnotationId.Image) {
-    annotations[`${annotationPrefix}/${SecretAnnotationId.Image}-0`] = annotation?.value;
+  if (secretType === SecretAnnotationId.Git) {
+    return `${annotationPrefix}/${SecretAnnotationId.Git}-${suffix}`;
+  }
+  if (secretType === SecretAnnotationId.Image) {
+    return `${annotationPrefix}/${SecretAnnotationId.Image}-${suffix}`;
   }
-  return annotations;
+  return null;
+};
+
+export const getSecretAnnotations = (
+  annotation: KeyValuePair,
+  existingAnnotations: { [key: string]: string } = {},
+) => {
+  let count = 0;
+  let annotationKey = getAnnotationKey(annotation?.key, count);
+  while (existingAnnotations[annotationKey]) {
+    annotationKey = getAnnotationKey(annotation?.key, ++count);
+  }
```

Comment:
Avoid duplicate entries

```suggestion
  while (existingAnnotations[annotationKey] && existingAnnotations[annotationKey] !== annotation?.value) {
    annotationKey = getAnnotationKey(annotation?.key, ++count);
  }
```

```  annotations:
    tekton.dev/git-0: 'https://github.com'
    tekton.dev/git-1: 'https://gitlab.com'
    tekton.dev/git-2: 'https://github.com'  // We should avoid this duplicate entries for the value
```

Reply:
Updated.